src/components/transactions/TxFilterForm/TxFilterForm.test.tsx

```diff
@@ -0,0 +1,247 @@
+
+    await act(async () => {
+      fireEvent.change(fromDateInput, { target: { value: '' } })
+      fireEvent.change(toDateInput, { target: { value: '' } })
+      fireEvent.change(toDateInput, { target: { value: toDate } })
+      fireEvent.change(fromDateInput, { target: { value: fromDate } })
+    })
+
+    expect(toDateInput).toHaveValue(toDate)
+    expect(fromDateInput).toHaveValue(fromDate)
+
+    expect(await screen.findByText(errorMsgStartDate, { selector: 'label' })).toBeInTheDocument()
+  })
+
+  it('Verify there is error when start and end date contain far future dates', async () => {
+    renderComponent()
+
+    const futureDate = '20/01/2036'
+    const errorMsgFutureDate = 'Date cannot be in the future'
+
+    const fromDateInput = screen.getAllByPlaceholderText(placeholder)[0]
+    const toDateInput = screen.getAllByPlaceholderText(placeholder)[1]
+
+    await act(async () => {
+      fireEvent.change(fromDateInput, { target: { value: fromDate } })
+      fireEvent.change(toDateInput, { target: { value: futureDate } })
+    })
+
+    expect(await screen.findByText(errorMsgFutureDate, { selector: 'label' })).toBeInTheDocument()
+
+    await act(async () => {
+      fireEvent.change(fromDateInput, { target: { value: futureDate } })
+      fireEvent.change(toDateInput, { target: { value: toDate } })
+    })
+
+    expect(await screen.findByText(errorMsgFutureDate, { selector: 'label' })).toBeInTheDocument()
+  })
+
+  it('Verify that when entering invalid characters in token filed shows an error message', async () => {
+    renderComponent()
+
+    const token = '694urt5'
+
+    const tokenInput =
+      screen.getByTestId('token-input').querySelector('input') ??
+      (() => {
+        throw new Error('Input element not found')
+      })()
+
+    expect(tokenInput).toBeInTheDocument()
+
+    await act(async () => {
+      fireEvent.change(tokenInput, { target: { value: token } })
+    })
+
+    expect(await screen.findByText(errorMsgFormat, { selector: 'label' })).toBeInTheDocument()
+  })
+
+  it('Verify there is error when 0 is entered in amount field', async () => {
+    renderComponent()
+
+    const errorMsgZero = 'The value must be greater than 0'
+    const amountInput =
+      screen.getByTestId('amount-input').querySelector('input') ??
+      (() => {
+        throw new Error('Input element not found')
+      })()
+
+    expect(amountInput).toBeInTheDocument()
+
+    await act(async () => {
+      fireEvent.change(amountInput, { target: { value: '0' } })
+    })
+
+    expect(await screen.findByText(errorMsgZero, { selector: 'label' })).toBeInTheDocument()
+  })
+
+  it('Verify that entering negative numbers and a non-numeric value in the amount filter is not allowed', async () => {
+    renderComponent()
+
+    const amountInput =
+      screen.getByTestId('amount-input').querySelector('input') ??
+      (() => {
+        throw new Error('Input element not found')
```

Comment:
why are you throwing here? On the next line you expect this element to be in the document?

Reply:
I had the error `Type 'null' is not assignable to type 'Element | Node | Document | Window'.ts(2345)` which was solved by adding that statement.

Reply:
that is strange

Reply:
changed, thank you for feedback